Dockerfile

```diff
@@ -1,6 +1,15 @@
 # hadolint ignore=DL3007
 FROM ghcr.io/opensafely-core/base-docker:latest
 
+# Some static metadata for this specific image, as defined by:
+# https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys
+# The org.opensafely.action label is used by the jobrunner to indicate this is
+# an approved action image to run.
+LABEL org.opencontainers.image.title="databuilder" \
+      org.opencontainers.image.description="Data Builder action for opensafely.org" \
+      org.opencontainers.image.source="https://github.com/opensafely-core/databuilder" \
+      org.opensafely.action="databuilder"
```

Comment:
Is this line definitely required? It was included in the `cohortextractor` image configuration that I link to in the commit message, so I've included it.

But I can't quite see where this is checked in job-runner.

There is this [`ALLOWED_IMAGES`](https://github.com/opensafely-core/job-runner/blob/4041d0d7442c8a385c23c9a6d9f0a37e1ceebc0c/jobrunner/config.py#L48-L56) configuration — but I'm not convinced this label is being checked. I could be mistaken.

Reply:
Good question. I suspect this might have been aspirational at the time it was written.

Reply:
Checked with @bloodearnest and he said we're not using this now, but could in future to:

* ditch the `ALLOWED_IMAGES` list and use this label to determine which images are allowed
* restrict `docker image prune` to these images in `opensafely-cli`

So I'll leave it.